binrw/src/strings.rs

```diff
@@ -40,6 +36,96 @@ pub struct NullString(
     pub Vec<u8>,
 );
 
+impl BinRead for NullString {
+    type Args = ();
+
+    fn read_options<R: Read + Seek>(
+        reader: &mut R,
+        options: &ReadOptions,
+        _: Self::Args,
+    ) -> BinResult<Self> {
+        let mut values = vec![];
+
+        loop {
+            let val = <u8>::read_options(reader, options, ())?;
+            if val == 0 {
+                return Ok(Self(values));
+            }
+            values.push(val);
+        }
+    }
+}
+
+impl BinWrite for NullString {
+    type Args = ();
+
+    fn write_options<W: Write + Seek>(
+        &self,
+        writer: &mut W,
+        options: &crate::WriteOptions,
+        args: Self::Args,
+    ) -> BinResult<()> {
+        self.0.write_options(writer, options, args)?;
+        0u8.write_options(writer, options, args)?;
+
+        Ok(())
+    }
+}
+
+impl From<&str> for NullString {
+    fn from(s: &str) -> Self {
+        Self(s.as_bytes().to_vec())
+    }
+}
+
+impl From<String> for NullString {
+    fn from(s: String) -> Self {
+        Self(s.into_bytes())
+    }
+}
+
+impl From<NullString> for Vec<u8> {
+    fn from(s: NullString) -> Self {
+        s.0
+    }
+}
+
+impl core::convert::TryFrom<NullString> for String {
+    type Error = FromUtf8Error;
+
+    fn try_from(value: NullString) -> Result<Self, Self::Error> {
+        String::from_utf8(value.0)
+    }
+}
+
+impl core::ops::Deref for NullString {
+    type Target = Vec<u8>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl core::ops::DerefMut for NullString {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+impl fmt::Debug for NullString {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "NullString(\"")?;
+        display_utf8(&self.0, f, |input| input.escape_debug())?;
```

Comment:
Yeah, when I am debugging trees of objects with the pretty printing output, I also find it more annoying than helpful when thin wrappers end up also receiving more whitespace and indentation, so ignoring that is maybe more of a feature than not? Though it is inconsistent. There would need to be a wrapper type either way to do the conversion since the inner is `Vec<u8>` so it isn’t like it can be sent as-is to debug_tuple. I have no strong opinions one way or the other.